storagemarket/dealstatus.go

```diff
@@ -65,6 +65,14 @@ const (
 	// StorageDealTransferring means data is being sent from the client to the provider via the data transfer module
 	StorageDealTransferring
 
+	// StorageDealProviderTransferRestart means a storage deal data transfer from client to provider will be restarted
```

Comment:
at this point, we should put these at the end of the states list -- people are starting to rely on the deal states list not changing